### ISSUE TITLE
Error wazuh web no open. 

### ISSUE THREAD

**Issue by jmchico** [wazuh/wazuh-kibana-app#1807]
Hi,
first thanks for developing this fantastic project. I think it's very very interesting!

I installed wazuh manager and api on single server, Then, I configured API connection but now when I Try to open wazuh in kibana the following error is shown:

 {"data":{"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred"},"status":500,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","headers":{"Accept":"application/json, text/plain, /","kbn-version":"6.2.3"},"timeout":8000,"url":"/api/wazuh-api/apiEntries"},"statusText":"Internal Server Error"}

![image](https://user-images.githubusercontent.com/55496386/65153555-3f7aba00-da2a-11e9-876b-8a66230f9a5a.png)

I'm not expert so could someone help me?
Thanks a lot! Really appreciate your help!

Version of api

{
   "error": 0,
   "data": {
      "msg": "Welcome to Wazuh HIDS API",
      "api_version": "v3.2.1",
      "hostname": "NPCI-SIEM",
      "timestamp": "Wed Sep 18 2019 15:38:33 GMT+0200 (CEST)"
   }
}

elasticsearch version 

{
  "name" : "hYDmCu_",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "pnonQD0dT-2eJrpSx7aD1A",
  "version" : {
    "number" : "6.2.4",
    "build_hash" : "ccec39f",
    "build_date" : "2018-04-12T20:37:28.497551Z",
    "build_snapshot" : false,
    "lucene_version" : "7.2.1",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  },
  "tagline" : "You Know, for Search"
}

Wazuh version

Wazuh v3.2.1 - Wazuh Inc.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (version 2) as
published by the Free Software Foundation. For more details, go to
https://www.gnu.org/licenses/gpl.html

Plugins 

{
  "name": "wazuh",
  "version": "3.2.1",
  "revision": "0387",
  "kibana": {
    "version": "6.2.3"
  },
  "description": "Wazuh App",
  "main": "index.js",
  "keywords": [
    "kibana",
    "wazuh",
    "ossec"
  ],
  "author": "Wazuh, Inc",
  "license": "GPL-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/wazuh/wazuh-kibana-app.git"
  },
  "bugs": {
    "url": "https://github.com/wazuh/wazuh-kibana-app/issues"
  },
  "homepage": "https://www.wazuh.com/",
  "dependencies": {
    "angular-animate": "1.6.5",
    "angular-aria": "1.6.5",
    "angular-cookies": "1.6.5",
    "angular-material": "1.1.1",
    "angular-md5": "^0.1.10",
    "ansicolors": "^0.3.2",
    "install": "^0.10.1",
    "js-yaml": "3.10.0",
    "lodash": "3.10.1",
    "needle": "^2.0.1",
    "node-cron": "^1.1.2",
    "winston": "3.0.0-rc1"
  }
}

**Comment by jsanchez91** [comment]
Hi jmchico,

Can you show me the result of this API method?

`http://<yourhost>:9200/_cluster/health/?level=shards`

Thanks

**Comment by jmchico** [comment]
Hi jose thanks for help.

It is only a server, not a cluster. This can affect the response. ?

because in the web that you indicate to me it puts ..../_cluster/...

This is de error 

ERR_CONNECTION_REFUSED

Thanks.

**Comment by jsanchez91** [comment]
Hi jmchico,

Elasticsearch install always as a cluster, if you only have one instance this is the `node-1`

This is my result of running `GET _cluster/health/?level=shards` in the Kibana Dev-Tools

![image](https://user-images.githubusercontent.com/3064506/65227890-bf0b9600-dac9-11e9-9a19-640e0c40105d.png)

**Comment by jmchico** [comment]
Hi 

The Kibana Dev-Tools does not work. 
No web application works.

The elasticsearch service restart now and no work. 
the same error comes out.  ERR_CONNECTION_REFUSED 

root@NPCI-SIEM:~# systemctl status elasticsearch.service
● elasticsearch.service - Elasticsearch
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch.service; enabled; vendor preset: enabled)
   Active: active (running) since jue 2019-09-19 10:55:33 CEST; 4min 9s ago
     Docs: http://www.elastic.co
 Main PID: 8550 (java)
    Tasks: 101
   Memory: 1.4G
      CPU: 4min 51.332s
   CGroup: /system.slice/elasticsearch.service
           └─8550 /usr/bin/java -Xms1g -Xmx1g -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnl

sep 19 10:55:33 NPCI-SIEM systemd[1]: Started Elasticsearch.

and if I launch the console  GET _cluster/health/?level=shards, 

this is the result

root@NPCI-SIEM:~# GET _cluster/health/?level=shards
Can't connect to _cluster:80

Nombre o servicio desconocido at /usr/share/perl5/LWP/Protocol/http.pm line 47.

**Comment by jsanchez91** [comment]
To get the cluster health in the command line use curl:

```bash
curl -XGET "http://<yourhost>:9200/_cluster/health/?level=shards"
```

Change `<yourhost>` for your hostname or IP, if you are in the command line of the server use `localhost` .

**Comment by jmchico** [comment]
root@NPCI-SIEM:~# curl -XGET "http://localhost:9200/_cluster/health/?level=shards" |more
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0{"cluster_name":"elasticsearch","status":"red","timed_out":false,"number_of_nodes":1,"number_of_data_nodes":1,"active_primary_shards":1044,"a
ctive_shards":1044,"relocating_shards":0,"initializing_shards":4,"unassigned_shards":2296,"delayed_unassigned_shards":0,"number_of_pending_ta
sks":64,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":22586,"active_shards_percent_as_number":31.220095693779903,"indices"
:{"wazuh-monitoring-3.x-2019.07.28":{"status":"yellow","number_of_shards":5,"number_of_replicas":1,"active_primary_shards":5,"active_shards":
5,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":5,"shards":{"0":{"status":"yellow","primary_active":true,"active_shards":
1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"1":{"status":"yellow","primary_active":true,"active_shards":1,"reloca
ting_shards":0,"initializing_shards":0,"unassigned_shards":1},"2":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shar
ds":0,"initializing_shards":0,"unassigned_shards":1},"3":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"in
itializing_shards":0,"unassigned_shards":1},"4":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializin
g_shards":0,"unassigned_shards":1}}},"wazuh-alerts-3.x-2019.07.15":{"status":"yellow","number_of_shards":5,"number_of_replicas":1,"active_pri
mary_shards":5,"active_shards":5,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":5,"shards":{"0":{"status":"yellow","primar
y_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"1":{"status":"yellow","primary_active"
:true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"2":{"status":"yellow","primary_active":true,"ac
tive_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"3":{"status":"yellow","primary_active":true,"active_shar
ds":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"4":{"status":"yellow","primary_active":true,"active_shards":1,"re
locating_shards":0,"initializing_shards":0,"unassigned_shards":1}}},"wazuh-alerts-3.x-2019.07.14":{"status":"yellow","number_of_shards":5,"nu
mber_of_replicas":1,"active_primary_shards":5,"active_shards":5,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":5,"shards":
{"0":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"1":{"st
atus":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"2":{"status":"ye
llow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"3":{"status":"yellow","pr
imary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"4":{"status":"yellow","primary_act
ive":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1}}},"wazuh-monitoring-3.x-2019.07.29":{"status
":"yellow","number_of_shards":5,"number_of_replicas":1,"active_primary_shards":5,"active_shards":5,"relocating_shards":0,"initializing_shards
":0,"unassigned_shards":5,"shards":{"0":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards
":0,"unassigned_shards":1},"1":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unas
signed_shards":1},"2":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_sh
ards":1},"3":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},
"4":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1}}},"wazuh-
alerts-3.x-2019.07.13":{"status":"yellow","number_of_shards":5,"number_of_replicas":1,"active_primary_shards":5,"active_shards":5,"relocating
_shards":0,"initializing_shards":0,"unassigned_shards":5,"shards":{"0":{"status":"yellow","primary_active":true,"active_shards":1,"relocating
_shards":0,"initializing_shards":0,"unassigned_shards":1},"1":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":
0,"initializing_shards":0,"unassigned_shards":1},"2":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initia
lizing_shards":0,"unassigned_shards":1},"3":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_sh
ards":0,"unassigned_shards":1},"4":{"status":"yellow","primary_active":true,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"
unassigned_shards":1}}},"wazuh-alerts-3.x-2019.07.12":{"status":"yellow","number_of_shards":5,"number_of_replicas":1,"active_primary_shards":
5,"active_shards":5,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":5,"shards":{"0":{"status":"yellow","primary_active":tru
e,"active_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"1":{"status":"yellow","primary_active":true,"active
_shards":1,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":1},"2":{"status":"yellow","primary_active":true,"active_shards":

**Comment by jsanchez91** [comment]
Your Elasticsearch cluster is in the red state
`"cluster_name":"elasticsearch","status":"red"`

In this blog post, explain how to identify the problem and solve it [RED Elasticsearch Cluster? Panic no longer](https://www.elastic.co/es/blog/red-elasticsearch-cluster-panic-no-longer)

To help you, I need the output of these requests:
```bash
curl -XGET "http://<yourhost>:9200/_cat/indices"
curl -XGET "http://<yourhost>:9200/_cluster/allocation/explain"
```

**Comment by jmchico** [comment]
root@NPCI-SIEM:~# curl -XGET "http://localhost:9200/_cluster/allocation/explain"
{"index":"wazuh-alerts-3.x-2019.07.05","shard":3,"primary":true,"current_state":"unassigned","unassigned_info":{"reason":"CLUSTER_RECOVERED","at":"2019-09-20T10:07:30.452Z","last_allocation_status":"throttled"},"can_allocate":"throttled","allocate_explanation":"allocation temporarily throttled","node_allocation_decisions":[{"node_id":"hYDmCu_-QqahEZDQmxqgXA","node_name":"hYDmCu_","transport_address":"127.0.0.1:9300","node_decision":"throttled","store":{"in_sync":true,"allocation_id":"RdNzlKNyQ7GzpAJHvD6S4g"},"deciders":[{"decider":"throttling","decision":"THROTTLE","explanation":"reached the limit of ongoing initial primary recoveries [4], cluster setting [cluster.routing.allocation.node_initial_primaries_recoveries=4]"}]}]}root@NPCI-SIEM:~#


root@NPCI-SIEM:~# curl -XGET "http://localhost:9200/_cat/indices"
red    open wazuh-alerts-3.x-2019.06.10     rl-Iy7NZQu-V78oT2A5ilQ 5 1
red    open wazuh-monitoring-3.x-2019.05.29 QLHdHbhwT5mLB7UbB9w5vg 5 1
red    open wazuh-alerts-3.x-2019.08.05     v_8zOhz3SbWFuT1GOrsL0Q 5 1
red    open .wazuh                          qUzvW66tSLyyyk1vSQhPoQ 1 1
red    open wazuh-alerts-3.x-2019.07.30     QhgVg0X2S1KDMljfBwrDaA 5 1
red    open wazuh-alerts-3.x-2019.08.19     Q6-oHfpeSDSWjhfaA1WREw 5 1
red    open wazuh-monitoring-3.x-2019.08.05 31XWHWEHRX-oUNQiRvwuDA 5 1
red    open wazuh-monitoring-3.x-2019.07.20 fuUaQ0TpS1WeFCjcPpXBzA 5 1
red    open wazuh-alerts-3.x-2018.05.23     lmcQ87lARHWt5e11C7wgfQ 5 1
red    open wazuh-alerts-3.x-2019.08.06     qda00_XXR6yMFFozYZDTbA 5 1
red    open wazuh-monitoring-3.x-2018.05.17 tMRNWn0ZQbGKxu7xt8tZJg 5 1
red    open wazuh-monitoring-3.x-2019.07.28 bntute9jQY-timOrmeayeg 5 1
red    open wazuh-monitoring-3.x-2019.06.12 y2_uQ2dUTJ-eVmX4DIT-OA 5 1
red    open wazuh-monitoring-3.x-2019.05.15 mqrdPbxKQZ-LX5LYN076qQ 5 1
yellow open wazuh-monitoring-3.x-2019.08.25 8rwp-Mo0QtKTkiULJ_a8SQ 5 1  1440 0 659.9kb 659.9kb
red    open wazuh-monitoring-3.x-2018.05.05 akPmFo5OSvmZ7f-Omy744Q 5 1
red    open wazuh-monitoring-3.x-2019.08.23 -r1XORjtR4O7y_k0-9Xe9g 5 1  1440 0   800kb   800kb
red    open wazuh-monitoring-3.x-2019.07.24 GnH5l2HoTD-q2vaRVUs5OQ 5 1
red    open wazuh-alerts-3.x-2019.05.18     sSuvzMnRQmqB6YFm3yOfgQ 5 1
red    open wazuh-alerts-3.x-2019.07.31     z787Vf9iSuq8K-DWy8p2Tw 5 1
red    open wazuh-monitoring-3.x-2019.08.11 Lg2N2tDfTWOhX9bTNWWFzw 5 1
red    open wazuh-alerts-3.x-2019.05.07     s1h5Qh7RRm68ie-yEDtLPA 5 1
red    open wazuh-alerts-3.x-2019.07.08     SV-vY1-CSsaH6Cofp_asbw 5 1
yellow open wazuh-monitoring-3.x-2019.09.08 rx6f_U7EQ4iWWfRifOKhxQ 5 1  1440 0   621kb   621kb

**Comment by jsanchez91** [comment]
Hi @jmchico 
This is the problem here: `"unassigned_shards":2296`

Let's force the shard allocation:
```bash
curl -XPUT '<yourhost>:9200/_cluster/settings' -H 'Content-Type: application/json' -d '{ "transient": { "cluster.routing.allocation.enable" : "all" } }'
```
This command take some time, execute the next command for about 5-10 minutes
```bash
watch -n0 "curl -s <yourhost>:9200/_cluster/health?pretty | grep active_shards_percent"
```
What we are doing is showing just active_shards_percent_as_number every 0,1 seconds, this way we can check if that number is changing or not. Let us know if that number is growing up or not. If it's growing up, your cluster is still recovering the shards.

Regards

**Comment by jmchico** [comment]
Hi jsanchez 

i try 

root@NPCI-SIEM:~# curl -XPUT 'localhost:9200/_cluster/settings' -H 'Content-Type: application/json' -d '{ "transient": { "cluster.routing.allocation.enable" : "all" } }'
curl: (7) Failed to connect to localhost port 9200: Conexión rehusada

and

root@NPCI-SIEM:~# curl -XPUT '10.133.3.11:9200/_cluster/settings' -H 'Content-Type: application/json' -d '{ "transient": { "cluster.routing.allocation.enable" : "all" } }'
curl: (7) Failed to connect to 10.133.3.11 port 9200: Conexión rehusada

No running.

**Comment by jsanchez91** [comment]
Hi jmchico,

Try to add `http://`

```bash
curl -XPUT 'http://localhost:9200/_cluster/settings' -H 'Content-Type: application/json' -d '{ "transient": {"cluster.routing.allocation.enable" : "all" } }'
```

If not work check your Elasticsearch service status:
```bash
systemctl status elasticsearch
```
Regards

**Comment by jmchico** [comment]
Hi Jose

my server has problems with elasticsearch.

It disconnects every 10 minutes.

The first case I thought it was a problem of space I extended it and we continued with the disconnections.


The error he gives me is:
root@NPCI-SIEM:~# service elasticsearch status
● elasticsearch.service - Elasticsearch
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since mar 2019-09-24 08:28:55 CEST; 29min ago
     Docs: http://www.elastic.co
  Process: 1171 ExecStart=/usr/share/elasticsearch/bin/elasticsearch -p ${PID_DIR}/elasticsearch.pid --quiet (code=exited, status=127)
 Main PID: 1171 (code=exited, status=127)

sep 24 08:16:42 NPCI-SIEM systemd[1]: Started Elasticsearch.
sep 24 08:28:55 NPCI-SIEM systemd[1]: elasticsearch.service: Main process exited, code=exited, status=127/n/a
sep 24 08:28:55 NPCI-SIEM systemd[1]: elasticsearch.service: Unit entered failed state.
sep 24 08:28:55 NPCI-SIEM systemd[1]: elasticsearch.service: Failed with result 'exit-code'.

Sorry, I could write to you in Spanish that I express myself better.

Regards.
